torch/csrc/jit/python/script_init.cpp

```diff
@@ -1090,6 +1090,22 @@ void initJitScriptBindings(PyObject* module) {
           })
       .def("apply", &Module::apply)
       .def("__copy__", &Module::copy)
+      .def(
+          "__hash__",
+          [](const Module& self) {
+            // Similar to Tensor's `__hash__`, which is `id()`.
+            return std::hash<c10::ivalue::Object*>{}(self._ivalue().get());
+          })
+      .def(
+          "__eq__",
+          [](const Module& self, const py::object& other) {
+            // TODO: call UDF if it exists
```

Comment:
Is there a way to easily raise an error if you hit the TODO case? Maybe not...

Reply:
I'm actually not sure if UDF `__eq__` is even a thing on `Module`. I could understand it for `Object` but not sure if it makes sense for `Module`

Speaking of which @gmagogsfm can we get a review on this from the TorchScript side? Thanks!